File: Napakalaki JAVA/src/napakalaki_java/Monster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki_java;

/**
 *
 * @author jumacasni
 */
public class Monster {
    private String name;
    private int combatLevel;
    private Prize prize;
    private BadConsequence badConsequence;
    private int levelChangeAgainstCultistPlayer;
    
    public Monster(String name, int level, BadConsequence bc, Prize prize){
        levelChangeAgainstCultistPlayer = 0;
        this.name = name;
        combatLevel = level;
        this.prize = prize;
        badConsequence = bc;
    }
    
    public Monster(String n, int l, BadConsequence badConsequence, Prize p, int lC){
        name = n;
        combatLevel = l;
        this.badConsequence = badConsequence;
        prize = p;
        levelChangeAgainstCultistPlayer = lC;
    }
    
    public String getName(){
        return name;
    }
    
    public int getCombatLevel(){
        return combatLevel;
    }
    
    public int getCombatLevelAgainstCultistPlayer(){
        return (combatLevel + levelChangeAgainstCultistPlayer);
    }
    
    public Prize getPrize(){
        return prize;
    }
    
    public BadConsequence getBadConsequence(){
        return badConsequence;
    }
    
    public int getLevelsGained(){
        return prize.getLevel();
    }
    
    public int getTreasuresGained(){
        return prize.getTreasures();
    }
    
    public String toString(){
        String s = "";
        s = s + name + "\nCombat Level = " + combatLevel + "\nBonus Sectarios = " + levelChangeAgainstCultistPlayer + "\nMal Rollo = "  + badConsequence.toString()
            + "\nBuen Rollo = " + prize.toString();
        return s;
    }
}

File: Napakalaki RUBY/lib/monster.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module NapakalakiGame
  class Monster
  
  def initialize(n, l, bc, p, lc = 0)
    @name = n
    @combatLevel = l
    @badConsequence = bc
    @prize = p
    @levelChangeAgainstCultistPlayer = lc
  end
  
  def getName()
    @name
  end
  
  def getCombatLevel()
    @combatLevel
  end
  
  def getBadConsequence()
    @badConsequence
  end
  
  def getPrize()
    @prize
  end
  
  def getLevelsGained()
    @prize.getLevel()
  end
  
  def getTreasuresGained()
    @prize.getTreasures()
  end
  
  def getCombatLevelAgainstCultistPlayer()
    @combatLevel + @levelChangeAgainstCultistPlayer
  end
  
  def to_s
    return "Nombre: #{@name} \n Nivel: #{@combatLevel} \n BadConsequence: #{@badConsequence.to_s}\n Prize: #{@prize.to_s}\n Bonus sectarios: #{@levelChangeAgainstCultistPlayer}\n"
  end
  
end
end


